### PR TITLE
Add new symbol for linux IP_MULTICAST_ALL

### LINKED ISSUE #1199: Support for IP_MULTICAST_ALL
I would like to request a friendly binding for the socket option IP_MULTICAST_ALL.

This was added in version 2.67 of [libc](https://github.com/rust-lang/libc), in the following [libc PR](https://github.com/rust-lang/libc/pull/1696/commits/b7cb4d2c7b988ca0ce82fc43ee6484035a578cf0)

### PR DESCRIPTION
**Pull request by anth1y**
this closes #1199 
/cc @asomers

### CODE REVIEW

File: src/sys/socket/sockopt.rs [hunk 1]
```diff
@@ -238,6 +238,17 @@ cfg_if! {
 }
 sockopt_impl!(Both, IpMulticastTtl, libc::IPPROTO_IP, libc::IP_MULTICAST_TTL, u8);
 sockopt_impl!(Both, IpMulticastLoop, libc::IPPROTO_IP, libc::IP_MULTICAST_LOOP, bool);
+cfg_if! {
```
Comment (asomers) [inline src/sys/socket/sockopt.rs:241]: You shouldn't use `cfg_if` if you're only going to have one clause.  Instead, just do
```rust
#[cfg(target_os = "linux")]
whatever
```

File: src/sys/socket/sockopt.rs [hunk 2]
```diff
@@ -238,6 +238,17 @@ cfg_if! {
...
+	 if #[cfg(any(target_os = "linux"))] {
```
Comment (asomers) [inline src/sys/socket/sockopt.rs:242]: The symbol is defined on Android, too.

**Review by asomers — CHANGES_REQUESTED** [review]
You also need a CHANGELOG entry.  And if possible, a test.  I don't know what this option does.  Is it practical to add a test for it in Nix?

**Comment by anth1y** [comment]
I dont think this needs a test but i'll defer to your judgement here:

```
       IP_MULTICAST_ALL (since Linux 2.6.31)
              This option can be used to modify the delivery policy of
              multicast messages to sockets bound to the wildcard
              INADDR_ANY address.  The argument is a boolean integer
              (defaults to 1).  If set to 1, the socket will receive
              messages from all the groups that have been joined
              globally on the whole system.  Otherwise, it will deliver
              messages only from the groups that have been explicitly
              joined (for example via the IP_ADD_MEMBERSHIP option) on
              this particular socket.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased] - ReleaseDate
 ### Added
+- Add new symbol for linux IP_MULTICAST_ALL
+  (#[1411](https://github.com/nix-rust/nix/pull/1411))
```
Comment (asomers) [inline CHANGELOG.md:9]: Looks like you copy/pasted the wrong PR number.

File: src/sys/socket/sockopt.rs (hunk 1 above)
Comment (asomers) [reply src/sys/socket/sockopt.rs:241]: Did you forget to push this change?

**Review by asomers — CHANGES_REQUESTED** [review]

**Review by asomers — COMMENTED** [review]
Better.  Can you please squash your commits?  Then we'll be good to go.

**Comment by anth1y** [comment]
going to close this my branch is super polluted ...

**Pull request closed without merging**